runtime/capabilitycontrollers_test.go

```diff
@@ -0,0 +1,771 @@
+                                    // Act
+                                    let gotCap: Capability<&AnyStruct>? =
+                                        %s.capabilities.get<&AnyStruct>(/public/r)
+
+                                    // Assert
+                                    assert(gotCap == nil)
+                                }
+                            }
+                        `,
+						accountExpression,
+					),
+				)
+				require.NoError(t, err)
+			})
+
+			t.Run("get and borrow existing, with valid type", func(t *testing.T) {
+
+				t.Parallel()
+
+				err, _, _ := test(
+					fmt.Sprintf(
+						// language=cadence
+						`
+                          import Test from 0x1
+
+                          transaction {
+                              prepare(signer: AuthAccount) {
+                                  let storagePath = /storage/r
+                                  let publicPath = /public/r
+                                  let expectedCapID: UInt64 = 1
+                                  let resourceID = 42
+
+                                  // Arrange
+                                  Test.createAndSaveR(id: resourceID, storagePath: storagePath)
+                                  let issuedCap: Capability<&Test.R> =
+                                      signer.capabilities.storage.issue<&Test.R>(storagePath)
+                                  signer.capabilities.publish(issuedCap, at: publicPath)
+
+                                  // Act
+                                  let gotCap: Capability<&Test.R> = %s.capabilities.get<&Test.R>(publicPath)!
+                                  let ref: &Test.R = gotCap.borrow()!
+
+                                  // Assert
+                                  assert(issuedCap.id == expectedCapID)
+                                  assert(gotCap.id == expectedCapID)
+                                  assert(ref.id == resourceID)
+                              }
+                          }
+                        `,
+						accountExpression,
+					),
+				)
+				require.NoError(t, err)
+			})
+
+			t.Run("get existing, with subtype", func(t *testing.T) {
+
+				t.Parallel()
+
+				err, _, _ := test(
+					fmt.Sprintf(
+						// language=cadence
+						`
+                          import Test from 0x1
+
+                          transaction {
+                              prepare(signer: AuthAccount) {
+                                  let storagePath = /storage/r
+                                  let publicPath = /public/r
+                                  let expectedCapID: UInt64 = 1
+                                  let resourceID = 42
+
+                                  // Arrange
+                                  Test.createAndSaveR(id: resourceID, storagePath: storagePath)
+                                  let issuedCap: Capability<&Test.R{}> =
+                                      signer.capabilities.storage.issue<&Test.R{}>(storagePath)
```

Comment:
Perhaps it is too late in the process for this feedback, but I have to say that this feels really unwieldy for what seems like a fairly common operation; we're going from a single method call on the `AuthAccount` to a triple-nested call.

Reply:
This is good feedback, but please leave it on the FLIP instead, where we discuss the design (I had the same concerns, BTW).